### ISSUE TITLE
Dehydration does not get initialized until the second log in

### ISSUE THREAD

**Issue by szykula** [vector-im/element-web#20333]
### Steps to reproduce

1. Add ```"features": {"feature_dehydration": true}``` to config.json
2. Create a new user account (Alice)
3. Log in creating a new session
4. Send Bob a message
5. Log out
6. \>>Bob responds<<
7. Log in again

### Outcome

The received message does not get decrypted: ```** Unable to decrypt: The sender's device has not sent us the keys for this message. **```. Any other conversations from this point onward get decrypted as intended - it seems that dehydration does not get initialized until the second log in.

### Application version

Element version 1.9.8, Olm version 3.2.8

### Will you send logs?

No

**Comment by szykula** [comment]
It seems that in the ```sign up → send a message → log out → sign in again``` scenario ```accessSecretStorage()``` gets executed only once, during the second log in (logs ```Setting dehydration key```). Please correct me if I'm mistaken, but doesn't that mean, that the dehydrated device gets created only when the storage needs to be accessed (i.e. not during the initial session and only after losing the initial keys)?

**Comment by Mikaela** [comment]
Duplicate of https://github.com/vector-im/element-web/issues/18977.
